File: src/Starter/Features/InternalSearchEngines/Settings/Views/Settings.axaml.fs
namespace Starter.Features.InternalSearchEngines.Settings.Views

open Starter.Features.InternalSearchEngines.Settings.Views
open Starter.Features.InternalSearchEngines.Settings.ViewModels

open Avalonia.Controls
open Avalonia.Markup.Xaml

open R3

type Settings() as this =
    inherit UserControl()

    do this.InitializeComponent()

    member this.InitializeComponent() =
        AvaloniaXamlLoader.Load this

        this.DataContextChanged.Add(fun _ ->
            match this.DataContext with
            | :? SettingsViewModel as vm ->
                let activatorPrefixes = this.GetControl<ItemsControl> "ActivatorPrefixes"

                // Set ActivatorPrefixes item template
                activatorPrefixes.ItemTemplate <- ActivatorPrefixes.searchEngineActivatorsItemTemplate this

                // Bind SearchEnginePrefixes source
                let sub = vm.SearchEngineActivators.Subscribe(fun vms -> activatorPrefixes.ItemsSource <- vms)

                this.Unloaded.Add(ignore >> sub.Dispose)
                this.Loaded.Add(fun _ -> vm.OnOpened())
            | _ -> ()
        )

File: src/Starter/Views/Settings.axaml.cs
using Avalonia.Controls;
using Avalonia.Interactivity;

namespace Starter.Views;

public partial class Settings : UserControl
{
    public Settings() => InitializeComponent();

    protected override void OnLoaded(RoutedEventArgs e)
    {
        base.OnLoaded(e);
        if (DataContext is not ViewModels.SettingsViewModel vm) return;
        vm.OnOpened();
    }
}
